### PR TITLE
feat: Add ability to disable prometheus scraping

### PR DESCRIPTION
**Pull request by TylerStaplerAtFanatics**
### Description

The current chart has no way of removing the `prometheus.io` annotations.
We use Datadog to ingest prometheus metrics, and I noticed that the metrics emitted by the aws-load-balancer-controller are very high cardinality, increasing our costs dramatically.

The datadog agent looks for applications with the annotation `prometheus.io/scrape: "true"`

### Tests of the new flag (truncated for brevity)
I'm using `yq` to test 
https://github.com/mikefarah/yq
```
❯ helm template aws-load-balancer-controller . --values ci/values.yaml --values values.yaml --set clusterName=testing --namespace test-ns | yq 'select(.kind=="Deployment") | .spec.template.metadata'
labels:
  app.kubernetes.io/name: aws-load-balancer-controller
  app.kubernetes.io/instance: aws-load-balancer-controller
annotations:
  prometheus.io/port: "8080"
  prometheus.io/scrape: "true"
```

```
❯ helm template aws-load-balancer-controller . --values ci/values.yaml --values values.yaml --set clusterName=testing --namespace test-ns --set prometheusScrape.enabled=false | yq 'select(.kind=="Deployment") | .spec.template.metadata'
labels:
  app.kubernetes.io/name: aws-load-balancer-controller
  app.kubernetes.io/instance: aws-load-balancer-controller
annotations: {}
```

```
❯ helm template aws-load-balancer-controller . --values ci/values.yaml --values values.yaml --set clusterName=testing --namespace test-ns --set prometheusScrape.enabled=true | yq 'select(.kind=="Deployment") | .spec.template.metadata'
labels:
  app.kubernetes.io/name: aws-load-balancer-controller
  app.kubernetes.io/instance: aws-load-balancer-controller
annotations:
  prometheus.io/port: "8080"
  prometheus.io/scrape: "true"
```

```
❯ helm template aws-load-balancer-controller . --values ci/values.yaml --values values.yaml --set clusterName=testing --namespace test-ns --set prometheusScrape.enabled=true --set serviceMonitor.enabled=true | yq 'select(.kind=="Deployment") | .spec.template.metadata'
labels:
  app.kubernetes.io/name: aws-load-balancer-controller
  app.kubernetes.io/instance: aws-load-balancer-controller
annotations: {}
```

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
         {{- toYaml .Values.podLabels | nindent 8 }}
         {{- end }}
       annotations:
-        {{- if not .Values.serviceMonitor.enabled }}
+        {{- if not .Values.serviceMonitor.enabled and .Values.prometheusScrape.enabled}}
```
Comment (mkilchhofer) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:30]: Wrong usage of helm `and` keyword.
Correct is
{{- if and (not .Values.serviceMonitor.enabled) .Values.prometheusScrape.enabled }}

**Comment by oliviassss** [comment]
@TylerStaplerAtFanatics, hey, would you address the comments? Thanks

**Comment by TylerStaplerAtFanatics** [comment]
👋  Sorry for the delay on this. We pursued a different mitigation internally and this fell off my radar.

I have some free time today and I'm working on refactoring this

File: helm/aws-load-balancer-controller/templates/deployment.yaml (hunk 1 above)
Comment (TylerStaplerAtFanatics) [reply helm/aws-load-balancer-controller/templates/deployment.yaml]: Thanks good catch

**Comment by TylerStaplerAtFanatics** [comment]
Hey @oliviassss! This should be ready for review, thanks for your patience.

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -27,13 +27,12 @@ spec:
         {{- toYaml .Values.podLabels | nindent 8 }}
         {{- end }}
       annotations:
-        {{- if not .Values.serviceMonitor.enabled }}
```
Comment (mkilchhofer) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:30]: Why so complicated?
IMO using something like I already suggested in comment https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3597#discussion_r1519577932 would be perfectly fine:

> Wrong usage of helm `and` keyword.
> Correct is
> `{{- if and (not .Values.serviceMonitor.enabled) .Values.prometheusScrape.enabled }}`

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 3]
```diff
@@ -27,13 +27,12 @@ spec:
...
-        prometheus.io/port: "{{ (split ":" .Values.metricsBindAddr)._1 | default 8080 }}"
-        {{- end}}
-        {{- if .Values.podAnnotations }}
-        {{- toYaml .Values.podAnnotations | nindent 8 }}
+        {{- $promethusAnnotations := dict }}
+        {{- if and .Values.prometheusScrape.enabled (not .Values.serviceMonitor.enabled) }}
+        {{- $promethusAnnotations = dict "prometheus.io/scrape" "true" "prometheus.io/port" "8080"}}
         {{- end }}
+        {{- $podAnnotations := merge $promethusAnnotations .Values.podAnnotations }}
+        {{- toYaml $podAnnotations | default (dict) | nindent 8 }}
```
Comment (mkilchhofer) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:35]: Why so complicated?
IMO using something like I already suggested in comment https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3597#discussion_r1519577932 would be perfectly fine:

> Wrong usage of helm `and` keyword.
> Correct is
> `{{- if and (not .Values.serviceMonitor.enabled) .Values.prometheusScrape.enabled }}`

**Pull request closed without merging**